cbrkit/sim/collections.py

```diff
@@ -26,3 +26,51 @@ def wrapped_func(x: Collection[Any], y: Collection[Any]) -> float:
         return dist2sim(jaccard_distance(x, y))
 
     return wrapped_func
+
+def smith_waterman(match_score: int = 2, mismatch_penalty: int = -1, gap_penalty: int = -1) -> SimPairFunc:
+    """
+    Performs the Smith-Waterman alignment with configurable scoring parameters. If no element matches it returns 0.0.
+
+    Args:
+        match_score (int, optional): Score for matching characters. Defaults to 2.
+        mismatch_penalty (int, optional): Penalty for mismatching characters. Defaults to -1.
+        gap_penalty (int, optional): Penalty for gaps. Defaults to -1.
+
+    Returns:
+        float: Alignment score for the two sequences.
+
+    Example:
+        >>> sim = smith_waterman()
+        >>> sim("abcde", "fghe")
+        2.0
+    """
+    from minineedle import smith, core
+
+    def wrapped_func(x: str, y: str) -> float:
+        try:
+            alignment = smith.SmithWaterman(x, y)
+            alignment.change_matrix(core.ScoreMatrix(match=match_score, miss=mismatch_penalty, gap=gap_penalty))
+            alignment.align()
+            return alignment.get_score()
+        except ZeroDivisionError:
+            return 0.0
+
+    return wrapped_func
+def dtw_similarity() -> SimPairFunc:
+    """Dynamic Time Warping similarity function.
+
+    Examples:
+        >>> sim = dtw_similarity()
+        >>> sim([1, 2, 3], [1, 2, 3, 4])
+        # Wert hängt von der konkreten Implementierung der Umrechnung von Distanz zu Ähnlichkeit ab
+    """
+    from dtaidistance import dtw
+    import numpy as np
+    def dist2sim(distance: float) -> float:
```

Comment:
There is already such a function in `cbrkit.helpers`. Is there a reason to add a new one?

Reply:
I actually havent seen that there is such a helper function, only difference is the missing check if its 0, will remove mine though.